[release 1.0] Sync upstream 2022-02-03

nunnatsa opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release notes**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
Sync upstream 2022-02-03
```

chenyosef commented:
/lgtm

bardielle commented on Dockerfile.rhel:
```diff
@@ -1,9 +1,27 @@
+        name="cluster-api-provider-kubevirt" \
+        architecture="x86_64" \
+        summary="cluster-api-provider-kubevirt" \
+        maintainer="Nir Argaman <nargaman@redhat.com>"
```
can we add multiple maintainers?

rgolangh commented:
I think so, but the best is to create a mailing list like ocprhv@redhat.com.


On Tue, 10 Nov 2020 at 12:20, Danielle Barda <notifications@github.com>
wrote:

> *@bardielle* commented on this pull request.
> ------------------------------
>
> In Dockerfile.rhel
> <https://github.com/openshift/cluster-api-provider-kubevirt/pull/13#discussion_r520448616>
> :
>
> > +#
> +#    http://www.apache.org/licenses/LICENSE-2.0
> +#
> +# Unless required by applicable law or agreed to in writing, software
> +# distributed under the License is distributed on an "AS IS" BASIS,
> +# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> +# See the License for the specific language governing permissions and
> +# limitations under the License.
> +
> +FROM registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.7 AS builder
> +
> +LABEL   com.redhat.component="machine-api" \
> +        name="cluster-api-provider-kubevirt" \
> +        architecture="x86_64" \
> +        summary="cluster-api-provider-kubevirt" \
> +        maintainer="Nir Argaman <nargaman@redhat.com>"
>
> can we add multiple maintainers?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/cluster-api-provider-kubevirt/pull/13#pullrequestreview-527054534>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABGBYHBVARWWAQGL6QQNJYTSPEHXDANCNFSM4TQNRY3Q>
> .
>

nirarg commented:
/lgtm

nirarg committed:
Change OWNERS file

agradouski committed:
expose cluster load balancer spec and metadata via KubevirtCluster resource

Signed-off-by: Alex Gradouski <agradouski@apple.com>

davidvossel committed:
Handle ssh keys properly for externally managed clusters

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add unit test exercising exernal managed clusters

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
E2e test to verify externally managed cluster

Signed-off-by: David Vossel <davidvossel@gmail.com>

nunnatsa committed:
Add go-linter to the github action

Fix all the golangci-linter warning (except for two deprecated function calls) and add the linter as a github action and a merge gate.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

nunnatsa committed:
Run goimports as part of the sanity test

run goimports to sort the imports in go files. goimports also performs gofmt.

To make it work, this commit also contains the sorted files as well.

This PR require a change in the way we're used to push PRs: it requires to make sure that the format of the files is valid before pushing the commits. For that, the PR also contain additions to the CONTRIBUTING.md file with the relevant instructions.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

rmohr committed:
Quote GOPROXY variable in the Makefile

Signed-off-by: Roman Mohr <rmohr@redhat.com>

k8s-ci-robot committed:
Merge pull request #86 from rmohr/test

Pass the GOPROXY variable properly to docker

k8s-ci-robot committed:
Merge pull request #87 from nunnatsa/golinter

Add go-linter to the github action

k8s-ci-robot committed:
Merge pull request #88 from nunnatsa/goimports

Sort the imports and reformat all golang files

davidvossel committed:
Ensure node provider id is set

Signed-off-by: David Vossel <davidvossel@gmail.com>

k8s-ci-robot committed:
Merge pull request #80 from agradouski/issue/64

expose cluster load balancer spec and metadata via KubevirtCluster resource

davidvossel committed:
Revise e2e test code style to conform to advanced gomega builtins

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
run make sanity

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add workaround for clustercli issue https://github.com/kubernetes-sigs/cluster-api/issues/6051

Signed-off-by: David Vossel <davidvossel@gmail.com>

davidvossel committed:
Add davidvossel to approvers list

Signed-off-by: David Vossel <davidvossel@gmail.com>

k8s-ci-robot committed:
Merge pull request #91 from davidvossel/append-owners

Add davidvossel to approvers list

k8s-ci-robot committed:
Merge pull request #83 from davidvossel/externally-managead-fix

Do not wait for SSH keys if there is an external controller

openshift-merge-robot committed:
Merge pull request #12 from nunnatsa/sync-upstream-2022-2-3

Sync upstream 2022-02-03

nirarg commented:
/lgtm
